### DISCUSSION TITLE
"Generating a new SSH key for a hardware security key" -> Neither prompt works...

### DISCUSSION THREAD

**Discussion by scottr99** [community/community#121092] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

General

### Body

I tried:
ssh-keygen -t ed25519-sk -C "myemailaddress@gmail.com"
and:
ssh-keygen -t ecdsa-sk -C "myemailaddress@gmail.com"

but for both commands I receive the following:

Generating public/private ecdsa-sk key pair.
You may need to touch your authenticator to authorize key generation.
No FIDO SecurityKeyProvider specified
Key enrollment failed: invalid format

Any help?

**Comment by scottr99** [comment]
The above is from the [following page](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key-for-a-hardware-security-key).

**Comment by ljste** [comment]
Hey scottr99,

It seems like the problem you're facing is with your system not recognizing your hardware security key properly. Here's what you can try:

Check Compatibility: Make sure your security key supports FIDO2 and works with the key types ed25519-sk or ecdsa-sk.

Install Necessary Software: Depending on your computer's operating system, you might need to install specific software or drivers to help your hardware security key communicate with your SSH client. For Windows users, this could mean getting OpenSSH compatible middleware for FIDO keys.

Update Your SSH Client: It's always a good idea to keep your SSH client up to date. Sometimes older versions don't fully support hardware security keys like FIDO/U2F ones.

Double-Check Your SSH Configuration: Take a look at your SSH configuration files to ensure there aren't any settings causing issues with key generation.

Hope these tips help you get it sorted!

**Reply by scottr99** [reply]
Hello, and thank you for your reply.

I wish what you have provided would help me, but unfortunately, it doesn't.

My hardware is a 1 y/o Mac w/ latest OS, so that shouldn't be the issue.

My SSH Client is up to date, and SSH Config. is good.

**Reply by ebndev** [reply]
Hey @ljstein99, thanks for wanting to contribute to the community!

Please note: We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. 

Thank you for helping us maintain an authentic and beneficial space for everyone.

**Comment by ZFPuhi** [comment, marked as answer]
In this case the **-sk** key types explicitly call for the presence of a hardware security key, but it doesn't sounds like you have added one.
Either add a FIDO2-compliant hardware key, or change the type argument to either **ecdsa** or **ed25519** (without the **-sk**).

**Reply by scottr99** [reply]
Thank you.

**Reply by ebndev** [reply]
@ZFPuhi thank you for all of your contributions to the Community in the last week! Appreciate you helping folks and answering questions 😁

**Comment by SrinivasBaligaZ** [comment]
Thanks @ZFPuhi -> you are on on Apr 25, 2024 helped me .
